clients/go/admin/client.go

```diff
@@ -48,19 +54,71 @@ func GetAdditionalAdminClientConfigOptions(cfg Config) []grpc.DialOption {
 	return opts
 }
 
-func NewAdminConnection(_ context.Context, cfg Config) (*grpc.ClientConn, error) {
+// This function assumes that the issuer supports the Authorization Server Metadata standard, and uses the oidc
+// library to retrieve the token endpoint.
+func getTokenEndpointFromIssuer(ctx context.Context, issuer string) (string, error) {
+	if issuer == "" {
+		logger.Errorf(ctx, "Attempting to construct provider with empty issuer")
+		return "", errors.New("cannot get token URL from empty issuer")
+	}
+
+	oidcCtx := oidc.ClientContext(ctx, &http.Client{})
+	provider, err := oidc.NewProvider(oidcCtx, issuer)
+	if err != nil {
+		logger.Errorf(ctx, "Error when constructing new OIDC Provider")
+		return "", err
+	}
+	logger.Infof(ctx, "Constructing Admin client with token endpoint %s", provider.Endpoint().TokenURL)
+
+	return provider.Endpoint().TokenURL, nil
+}
+
+// This retrieves a DialOption that contains a source for generating JWTs for authentication with Flyte Admin
+func getAuthenticationDialOption(ctx context.Context, cfg Config) (grpc.DialOption, error) {
+	var tokenURL string
+	tokenURL, err := getTokenEndpointFromIssuer(ctx, cfg.IssuerURL)
+	if err != nil || tokenURL == "" {
```

Comment:
to get this straight, even if getTokenEndpointFromIssuer returns an err that's fine if we can get the token url from the config?

Reply:
yeah.  is that bad?

Reply:
I added a comment, pushed.  Also renamed the function and the config option from `IssuerURL` to `AuthorizationServerURL` which I think is more accurate.